### PR TITLE
Expressions

### PR DESCRIPTION
**Pull request by hwbllmnn**
Replaces #400

### CODE REVIEW

File: .github/workflows/on-pull-request.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [12.x, 14.x, 16.x]
+        node-version: [16.x, 18.x]
```
Comment (KaiVolland) [inline .github/workflows/on-pull-request.yml:11]: IMHO we should support all node 14 to.

Comment (hwbllmnn) [reply .github/workflows/on-pull-request.yml]: It's reverted, we'll see if the workflows pass

Comment (hwbllmnn) [reply .github/workflows/on-pull-request.yml:9]: Unfortunately the workflow fails with node 12 and 14 for an unknown reason, ok to remove it?

File: src/SymbologyEncoder.ts [hunk 2]
```diff
@@ -65,14 +65,24 @@ export class SymbologyEncoder {
...
           } else if (!isFilterElement) {
             // rename all other tags to include the "se" namespace
             delete Object.assign(rule, {['se:' + key]: rule[key]})[key];
           }
           // recursive call ourselves
           if (val instanceof Object) {
             setTags([val], isFilterElement);
+          // } else {
+          //   console.log("asfdasdfasdfa");
+          //   console.log(val);
```
Comment (jansule) [inline src/SymbologyEncoder.ts:85]: This can probably be removed, right? :smile:

Comment (hwbllmnn) [reply src/SymbologyEncoder.ts]: Right :sweat_smile:

File: .github/workflows/on-pull-request.yml (hunk 1 above)
Comment (KaiVolland) [reply .github/workflows/on-pull-request.yml:9]: I'd prefer to have a node 14 run to. But as it's just about the build, it's not that important. So feel free to merge.

File: package.json [hunk 3]
```diff
@@ -32,20 +32,21 @@
...
     "core-js": "^3.22.4",
-    "geostyler-style": "^5.0.2",
+    "events": "^3.3.0",
+    "geostyler-style": "^6.0.0",
     "lodash": "^4.17.21",
     "stream": "^0.0.2",
     "string_decoder": "^1.3.0",
     "timers": "^0.1.1",
-    "xml2js": "^0.4.23"
+    "xml2js": "jansule/node-xml2js#d2f8280924618bd09c606f039cda477635de109f"
```
[the remark this answers is not in the document]
Comment (KaiVolland) [reply package.json:49]: We could go with your fork for know and switch to handmade SLD later. :thinking:

[the remark this answers is not in the document]
Comment (jansule) [reply package.json:49]: Yes, we definitely can do this. For better usage, I moved my fork to the geostyler organisation and published our changes. See https://www.npmjs.com/package/@geostyler/xml2js and https://github.com/geostyler/node-xml2js.

Please note the `@geostyler` namespace

[the remark this answers is not in the document]
Comment (hwbllmnn) [reply package.json]: Done in 8687909

File: src/SldStyleParser.v1.0.spec.ts [hunk 4]
```diff
@@ -795,6 +796,14 @@ describe('SldStyleParser implements StyleParser', () => {
       const { output: readStyle} = await styleParser.readStyle(sldString);
       expect(readStyle).toEqual(point_simplepoint_nestedLogicalFilters);
     });
+    it('can write a SLD style with expressions', async () => {
+      const { output: sldString } = await styleParser.writeStyle(polygonExpression);
+      expect(sldString).toBeDefined();
+      // As string comparison between two XML-Strings is awkward and nonsens
```
Comment (jansule) [inline src/SldStyleParser.v1.0.spec.ts:802]:
```suggestion
      // As string comparison between two XML-Strings is awkward and nonsense
```

**Review by jansule — APPROVED** [review]
Looks good to me now. Thanks @hwbllmnn! Just a non-blocking typo

**Pull request merged**